s3api/handlers_website.go

```diff
@@ -0,0 +1,212 @@
+		msg := fmt.Sprintf("failed building default website bucket access policy for %s: %s", bucketName, err.Error())
+		handleError(w, apierror.New(apierror.ErrInternalError, msg, err))
+		return
+	}
+
+	err = s3Service.UpdateBucketPolicy(r.Context(), &s3.PutBucketPolicyInput{
+		Bucket: aws.String(bucketName),
+		Policy: aws.String(string(defaultWebsitePolicy)),
+	})
+
+	// build the default IAM bucket admin policy (from the config and known inputs)
+	defaultPolicy, err := iamService.DefaultBucketAdminPolicy(aws.String(bucketName))
+	if err != nil {
+		msg := fmt.Sprintf("failed building default IAM policy for bucket %s: %s", bucketName, err.Error())
+		handleError(w, apierror.New(apierror.ErrInternalError, msg, err))
+		return
+	}
+
+	policyOutput, err := iamService.CreatePolicy(r.Context(), &iam.CreatePolicyInput{
+		Description:    aws.String(fmt.Sprintf("Admin policy for %s bucket", bucketName)),
+		PolicyDocument: aws.String(string(defaultPolicy)),
+		PolicyName:     aws.String(fmt.Sprintf("%s-BktAdmPlc", bucketName)),
+	})
+
+	if err != nil {
+		msg := fmt.Sprintf("failed to create policy: %s", err.Error())
+		handleError(w, errors.Wrap(err, msg))
+		return
+	}
+
+	// append policy delete to rollback tasks
+	rbfunc = func() error {
+		return func() error {
+			if _, err := iamService.DeletePolicy(r.Context(), &iam.DeletePolicyInput{PolicyArn: policyOutput.Policy.Arn}); err != nil {
+				return err
+			}
+			return nil
+		}()
+	}
+	rollBackTasks = append(rollBackTasks, rbfunc)
+
+	groupName := fmt.Sprintf("%s-BktAdmGrp", bucketName)
+	group, err := iamService.CreateGroup(r.Context(), &iam.CreateGroupInput{
+		GroupName: aws.String(groupName),
+	})
+
+	if err != nil {
+		msg := fmt.Sprintf("failed to create group: %s", err.Error())
+		handleError(w, errors.Wrap(err, msg))
+		return
+	}
+
+	// append policy delete to rollback tasks
+	rbfunc = func() error {
+		return func() error {
+			if _, err := iamService.DeleteGroup(r.Context(), &iam.DeleteGroupInput{GroupName: aws.String(groupName)}); err != nil {
+				return err
+			}
+			return nil
+		}()
+	}
+	rollBackTasks = append(rollBackTasks, rbfunc)
+
+	if _, err = iamService.AttachGroupPolicy(r.Context(), &iam.AttachGroupPolicyInput{
+		GroupName: aws.String(groupName),
+		PolicyArn: policyOutput.Policy.Arn,
+	}); err != nil {
+		msg := fmt.Sprintf("failed to create group: %s", err.Error())
+		handleError(w, errors.Wrap(err, msg))
+		return
+	}
+
+	output := struct {
+		Bucket *string
+		Policy *iam.Policy
+		Group  *iam.Group
+	}{
+		bucketOutput.Location,
+		policyOutput.Policy,
+		group.Group,
+	}
+
+	j, err := json.Marshal(output)
+	if err != nil {
+		log.Errorf("cannot marshal reasponse(%v) into JSON: %s", output, err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+	w.Write(j)
+}
+
+// rollBack executes functions from a stack of rollback functions
+func rollBack(t *[]func() error) {
```

Comment:
This will be shared, right? Can probably move to another file (`server.go`)

Reply:
yeah definately, when i do the reimplementation of the rollback for buckets, ill move it